File: code/java/dynamic_programming_article-03.java
public class Dynamic_programming_article_03 {
    /**
     * Classic dynamic programming: game problem
     */
    public Object solve(Object... args) {
        if (args == null || args.length != 1) {
            throw new IllegalArgumentException("Expected: (int[] piles)");
        }

        Solution solution = new Solution();
        return solution.stoneGame((int[]) args[0]);
    }

    private static class Solution {

        public boolean stoneGame(int[] piles) {
            return true;
        }
    }
}

File: code/python/dynamic_programming_article-03.py
"""Classic dynamic programming: game problem"""

from __future__ import annotations
from typing import Any


def solve(*args: Any, **kwargs: Any) -> Any:
    """Run this lesson exercise with positional inputs."""
    return _Solution().stoneGame(*args, **kwargs)


class _Solution:
    def stoneGame(self, piles: List[int]) -> bool:
        return True
